### PR TITLE
CMR-9190: Add paging to /access-control/permissions endpoint

### PR DESCRIPTION
**Pull request by daniel-zamora**

### CODE REVIEW

File: access-control-app/int-test/cmr/access_control/int_test/permission_check_test.clj [hunk 1]
```diff
@@ -35,16 +35,64 @@
...
+
+          "non int page_num"
+          bad-page-num2
+          "{\"errors\":[\"page_num must be a number greater than or equal to 1\"]}"
+
+          "non int page_size"
+          bad-page-size2
+          "{\"errors\":[\"page_size must be a number between 0 and 2000\"]}"
+
+
```
Comment (daniel-zamora) [inline access-control-app/int-test/cmr/access_control/int_test/permission_check_test.clj:77]: Space

File: access-control-app/int-test/cmr/access_control/int_test/permission_check_test.clj [hunk 2]
```diff
@@ -8,7 +8,7 @@
    [clojure.test :refer :all]
    [cmr.access-control.int-test.fixtures :as fixtures]
    [cmr.access-control.test.util :as u]
-   [cmr.common.util :refer [are3]]
+   [cmr.common.util :refer [are3 dissoc-in]]
```
Comment (daniel-zamora) [inline access-control-app/int-test/cmr/access_control/int_test/permission_check_test.clj:11]: remove dissoc-in

File: access-control-app/src/cmr/access_control/api/routes.clj [hunk 3]
```diff
@@ -163,12 +164,60 @@
...
+      (subvec vector start-index end-index))
+    vector))
+
+(defn get-permissions
+  "Formats a response for retrieving permissions based on the given parameters. When concept_id is present,
+  only a chunk of the concept_id vector will be processed according to page_size and page_num."
+  [context params headers]
+  (-> params
+      (select-keys [:page_size :page_num :concept_id])
+      util/map-keys->kebab-case
```
Comment (daniel-zamora) [inline access-control-app/src/cmr/access_control/api/routes.clj:183]: kebab-case twice

File: access-control-app/src/cmr/access_control/api/routes.clj [hunk 4]
```diff
@@ -370,12 +419,12 @@
         (OPTIONS "/" [] (common-routes/options-response))
 
         (GET "/"
-             {ctx :request-context params :params}
-             (get-permissions ctx params))
+             {ctx :request-context headers :headers params :params}
+             (get-permissions ctx params headers))
```
Comment (daniel-zamora) [inline access-control-app/src/cmr/access_control/api/routes.clj:423]: remove headers

**Review by eudoroolivares2016 — APPROVED** [review]

File: access-control-app/src/cmr/access_control/services/acl_service.clj [hunk 5]
```diff
@@ -386,7 +386,7 @@
           (for [group-id target-group-ids]
             [group-id (group-permissions-granted-by-acls group-id sids acls)]))))
 
-(defn-timed get-permissions
+(defn get-permissions
```
[the remark this answers is not in the document]
Comment (daniel-zamora) [reply access-control-app/src/cmr/access_control/services/acl_service.clj:389]:
```(defn-timed get-catalog-item-permissions
```

**Review by dpesall — APPROVED** [review]

**Pull request merged**